go-controller/pkg/ovn/multipolicy_test.go

```diff
@@ -630,12 +644,52 @@ var _ = ginkgo.Describe("OVN MultiNetworkPolicy Operations", func() {
 				err := app.Run([]string{app.Name})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
 			},
-			ginkgo.Entry("on local zone for layer3 topology", ovntypes.Layer3Topology, false),
-			ginkgo.Entry("on remote zone for layer3 topology", ovntypes.Layer3Topology, true),
-			ginkgo.Entry("on local zone for layer2 topology", ovntypes.Layer2Topology, false),
-			ginkgo.Entry("on remote zone for layer2 topology", ovntypes.Layer2Topology, true),
-			ginkgo.Entry("on local zone for localnet topology", ovntypes.LocalnetTopology, false),
-			ginkgo.Entry("on remote zone for localnet topology", ovntypes.LocalnetTopology, true),
+			NewTestCase(ovntypes.Layer3Topology).Build(),
+			NewTestCase(ovntypes.Layer3Topology).WithRemote().Build(),
+			NewTestCase(ovntypes.Layer2Topology).Build(),
+			NewTestCase(ovntypes.Layer2Topology).WithRemote().Build(),
+			NewTestCase(ovntypes.LocalnetTopology).Build(),
+			NewTestCase(ovntypes.LocalnetTopology).WithRemote().WithOVNMapping("physnet:breth0,network1:ovsbr1").Build(),
+			NewTestCase(ovntypes.LocalnetTopology).WithRemote().ExpectFailure(
+				`Warning ErrorReconcilingPod failed to find bridge mapping for network: "network1", node "node1", ovn-bridge-mappings: ""`,
+			).Build(),
 		)
 	})
 })
+
+type TestCaseBuilder struct {
+	topology      string
+	remote        bool
+	expectedError string
+	mapping       string
+}
+
+func NewTestCase(topology string) *TestCaseBuilder {
+	return &TestCaseBuilder{topology: topology}
+}
+
+func (b *TestCaseBuilder) WithRemote() *TestCaseBuilder {
+	b.remote = true
+	return b
+}
+
+func (b *TestCaseBuilder) ExpectFailure(msg string) *TestCaseBuilder {
+	b.expectedError = msg
+	return b
+}
+
+func (b *TestCaseBuilder) Build() ginkgo.TableEntry {
```

Comment:
```suggestion
func (b *TestCaseBuilder) Succeed() ginkgo.TableEntry {
```

Or `Success`. Something to indicate it "worked as expected".

Reply:
this func "builds" the entry
```
return ginkgo.Entry(
		fmt.Sprintf("on %s zone for %s topology", zone, b.topology), b.topology, b.remote, b.mapping, b.expectedError,
	)
```
one of the lines should fail, so Succeed might be weird ? (also since it just render, it makes build to sounds weird imho)
anyhow whatever you prefer

one option is to create `Failed` that equal this func, and then we can rename it to Succeed
and use `Failed` for the one expects failure

Reply:
So, imho, you should also use `build` to build the entry that expects failure, right ?

Reply:
i do aint it ? just on sep line
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5000/files#diff-6866db9fa9a4045b3db6ed1e842dc2a1c90a1565581a2e8db5f0c36adfa6e943R655

Reply:
:facepalm: 

Right, sorry. 

I had missed the `Build` on the separate line, my bad. You can disregard this conversation.